tests/suite/test_transport_server_udp_load_balance.py

```diff
@@ -0,0 +1,186 @@
+            ingress_controller_prerequisites.namespace
+        )
+
+        pattern = 'server .*;'
+        servers = re.findall(pattern, result_conf)
+        for key in endpoints.keys():
+            found = False
+            for server in servers:
+                if key in server:
+                    found = True
+            assert found
+
+    def test_udp_request_load_balanced_wrong_port(
+            self, kube_apis, crd_ingress_controller, transport_server_setup
+    ):
+        """
+        Requests to the load balanced UDP service should result in responses from 3 different endpoints.
+        """
+
+        patch_src = f"{TEST_DATA}/transport-server-udp-load-balance/wrong-port-transport-server.yaml"
+        patch_ts(
+            kube_apis.custom_objects,
+            transport_server_setup.name,
+            patch_src,
+            transport_server_setup.namespace,
+        )
+
+        wait_before_test()
+
+        port = transport_server_setup.public_endpoint.udp_server_port
+        host = transport_server_setup.public_endpoint.public_ip
+
+        print(f"sending udp requests to: {host}:{port}")
+        for i in range(3):
+            client = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0)
+            client.settimeout(2)
+            client.sendto("ping".encode('utf-8'), (host, port))
+            try:
+                client.recvfrom(4096)
+                # it should timeout, it's an error if it doesn't.
+                assert False
+            except socket.timeout:
+                print("successfully timed out")
+            client.close()
+
+        self.restore_ts(kube_apis, transport_server_setup)
+
+    def test_udp_request_load_balanced_missing_service(
+            self, kube_apis, crd_ingress_controller, transport_server_setup
+    ):
+        """
+        Requests to the load balanced UDP service should result in responses from 3 different endpoints.
+        """
+
+        patch_src = f"{TEST_DATA}/transport-server-udp-load-balance/missing-service-transport-server.yaml"
+        patch_ts(
+            kube_apis.custom_objects,
+            transport_server_setup.name,
+            patch_src,
+            transport_server_setup.namespace,
+        )
+
+        wait_before_test()
+
+        port = transport_server_setup.public_endpoint.udp_server_port
+        host = transport_server_setup.public_endpoint.public_ip
+
+        print(f"sending udp requests to: {host}:{port}")
+        for i in range(3):
+            client = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0)
+            client.settimeout(2)
+            client.sendto("ping".encode('utf-8'), (host, port))
+            try:
+                client.recvfrom(4096)
+                # it should timeout
+                assert False
+            except socket.timeout:
+                print("successfully timed out")
+            client.close()
+
+        self.restore_ts(kube_apis, transport_server_setup)
```

Comment:
we should add a test for listener port collision similar to tcp one.

Reply:
fixed